src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -20,20 +23,30 @@ namespace Temporalio.Worker
     /// <summary>
     /// Instance of a workflow execution.
     /// </summary>
-    internal class WorkflowInstance : TaskScheduler, IWorkflowInstance
+    internal class WorkflowInstance : TaskScheduler, IWorkflowInstance, IWorkflowContext
     {
         private readonly TaskFactory taskFactory;
-        private readonly WorkflowInstanceDetails details;
+        private readonly WorkflowDefinition defn;
+        private readonly IPayloadConverter payloadConverter;
+        private readonly IFailureConverter failureConverter;
+        private readonly Lazy<WorkflowInboundInterceptor> inbound;
+        private readonly Lazy<WorkflowOutboundInterceptor> outbound;
+        // Lazily created if asked for by user
+        private readonly Lazy<NotifyOnSetDictionary<string, WorkflowQueryDefinition>> mutableQueries;
+        // Lazily created if asked for by user
+        private readonly Lazy<NotifyOnSetDictionary<string, WorkflowSignalDefinition>> mutableSignals;
```

Comment:
Is the lazyness worth it?
What does it buy us? Is it just a general good practice?

Reply:
Last PR you dinged me on unnecessarily storing data when not needed, so this is kinda the same thing. Only users that ever request the `Workflow.Signals` property (so basically never) would create this.